test/json_client_test.rb

```diff
@@ -789,4 +789,20 @@ def test_should_raise_error_if_attempting_to_disable_timeout
       GdsApi::JsonClient.new(:timeout => -1)
     end
   end
+
+  def test_should_add_user_agent_using_env
+    previous_govuk_app_name = ENV['GOVUK_APP_NAME']
+    ENV['GOVUK_APP_NAME'] = "api-tests"
+
+    url = "http://some.other.endpoint/some.json"
+    stub_request(:get, url).to_return(:status => 200)
+
+    GdsApi::JsonClient.new.get_json(url)
+
+    assert_requested(:get, %r{/some.json}) do |request|
+      request.headers["User-Agent"] == "gds-api-adapters/#{GdsApi::VERSION} (api-tests)"
+    end
+
+    ENV['GOVUK_APP_NAME'] = previous_govuk_app_name
```

Comment:
If this test fails, this value won't get reset, and therefore could trigger a failure in other tests.  It would be good to reset this in an `ensure` clause just to be sure.